keras_nlp/src/models/resnet/resnet_backbone.py

```diff
@@ -0,0 +1,540 @@
+# Copyright 2024 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import keras
+from keras import layers
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.backbone import Backbone
+from keras_nlp.src.utils.keras_utils import standardize_data_format
+
+
+@keras_nlp_export("keras_nlp.models.ResNetBackbone")
+class ResNetBackbone(Backbone):
```

Comment:
Do we think this makes more sense as one architecture or two? We can do both here. Either `ResNetBackbone` and `ResNet2Backbone` or the one like we have here.

Most of the args makes sense here, but `block_type: "block"` vs `block_type: "basic_block"` is a little hard to understand. But also night to have less duplication.

Basically it's on us to define what constitutes a new architecture.

Reply:
how about providing different architectures as main backbone subclasses with built in configs?

Reply:
Talked offline. Let's just stick with the one class for now, nice and simple.

If we can think of something still short that's a little more descriptive than `block_type: "block"` and `block_type: "basic_block"` let's go for it, but I think this is fine as is.

Reply:
Updated to `block_type: "bottleneck_block"` and `block_type: "basic_block"`.
This should be consistent with the original paper: https://arxiv.org/abs/1512.03385